spec.bs

```diff
@@ -364,15 +453,28 @@ To get the <dfn>max batch size</dfn> for an [=/origin=] |issuer|, run the follow
 
 To <dfn>generate masked tokens</dfn> given [=key commitments=] |issuerKeys| and number |numTokens|, run the following steps:
 
-1. Let |maskOperation| be the Mask operation corresponding to |issuerKeys|["protocol_version"].
-1. Let |result| ([=tuple=] ([=byte sequence=], [=byte sequence=])) be the result of running |maskOperation| on |issuerKeys| and |numTokens|.
-1. If |result| is null, return an error.
-1. Return |result|.
-
-To <dfn>unmask tokens</dfn> given [=key commitments=] |issuerKeys|, byte string |pretokens|, and a bytestring |response|, run the following steps:
-
-1. Let |unmaskOperation| be the Unmask operation corresponding to |issuerKeys|["protocol_version"].
-1. Let |result| ([=list=] of byte strings) be the result of running |unmaskOperation| on |pretokens| and |response|.
+1. Let |request| be a [=byte sequence=] consisting of |numTokens| encoded as a uint16.
+1. Let |blinds| be an empty [=list=].
+1. Repeat the following steps, |numTokens| times:
+    1. Let |input| be an [=implementation-defined=] random value.
+    1. Let |result| ([=tuple=] ([=byte sequence=], [=byte sequence=])) be the result of running the **Blind** VOPRF function from [[!VOPRF]], where the second element is encoded as a X9.62 uncompressed point.
+    1. [=list/Append=] the first element of |result| to |blinds|.
+    1. Append the second element of |result| to |request|.
+1. Return [=tuple=] (|request|, |blinds|).
+
+To <dfn>unmask tokens</dfn> given [=key commitments=] |issuerKeys|, byte string |blinds|, and a bytestring |response|, run the following steps:
+
+1. Let |input| be empty [=byte sequence=].
+1. Let |evaluatedElements| be an empty [=list=].
+1. Let |blindedElements| be an empty [=list=].
+1. Let |issueResponse| be the deserialization of |response| as an **IssueResponse**.
+1. [=list/For each=] |nonce| of |issueResponse|["signed"]:
+    1. [=list/Append=] |nonce|["blinded"] to |blindedElements|.
+    1. [=list/Append=] |nonce|["evaluated"] to |evaluatedElements|.
+1. Let |pkS| be |issuerKeys|["keys"][|issueResponse|["key_id"]]["Y"].
+1. Let |proof| be |issueResponse|["proof"].
+1. Let |result| ([=list=] of byte strings) be the result of running **PSTFinalize** on |input|, |blinds|, |evaluatedElements|, |blindedElements|, |pkS|, and |proof|.
+1. If |result| is null, return "{{InvalidStateError}}" {{DOMException}.
```

Comment:
If I'm not mistaken this runs during fetch so there's no point returning (or even throwing) a DOMException here. Looking at https://wicg.github.io/trust-token-api/#issue-response it seems to be already handling null responses by returning a network error so just returning null here seems fine?

Reply:
As a side note Handling Issue Responses has some inconsistent return behavior that makes me question whether it does the right thing in all situations. Might be worth double-checking.

Reply:
Fixed up the inconsistent nulls and removed the error handling here.